lib/cli-manager.js

```diff
@@ -0,0 +1,352 @@
+/**
+ * Copyright (c) 2017, Kinvey, Inc. All rights reserved.
+ *
+ * This software is licensed to you under the Kinvey terms of service located at
+ * http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+ * software, you hereby accept such terms of service  (and any agreement referenced
+ * therein) and agree that you have read, understand and agree to be bound by such
+ * terms of service and are of legal age to agree to such terms with Kinvey.
+ *
+ * This software contains valuable confidential and proprietary information of
+ * KINVEY, INC and is subject to applicable licensing agreements.
+ * Unauthorized reproduction, transmission or distribution of this file and its
+ * contents is a violation of applicable laws.
+ */
+
+const async = require('async');
+const chalk = require('chalk');
+const yargs = require('yargs');
+
+const request = require('request');
+
+const { AllCommandsNotRequiringAuth, AuthOptionsNames, EnvironmentVariables, Errors, LogLevel } = require('./constants');
+const Authentication = require('./authentication');
+const KinveyError = require('./kinvey-error');
+const Request = require('./request');
+const utils = require('./utils');
+const cmdInit = require('./commands/init');
+const InitController = require('./controllers/init-controller');
+const cmdFlex = require('./commands/flex');
+const FlexController = require('./controllers/flex-controller');
+const FlexService = require('./services/flex-service');
+
+
+// TODO: refactor the whole setup concept
```

Comment:
I think it's okay to leave TODO statements in the codebase, but if possible, add some context - i.e. add why the setup concept needs to be refactored